woocommerce-services.php

```diff
@@ -1234,23 +1234,23 @@ public function add_tracking_info_to_emails( $order, $sent_to_admin, $plain_text
 
 			if ( $plain_text ) {
 				echo "\n=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n\n";
-				echo mb_strtoupper( __( 'Tracking', 'woocommerce-services' ), 'UTF-8' ) . "\n\n";
-				echo $markup;
+				echo esc_html(mb_strtoupper( __( 'Tracking', 'woocommerce-services' ), 'UTF-8' )) . "\n\n";
+				echo $markup; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped 
 				return;
 			}
 
 			?>
 				<div style="font-family: 'Helvetica Neue', Helvetica, Roboto, Arial, sans-serif; margin-bottom: 40px;">
-					<h2><?php echo __( 'Tracking', 'woocommerce-services' ); ?></h2>
+					<h2><?php esc_html_e( 'Tracking', 'woocommerce-services' ); ?></h2>
 					<table class="td" cellspacing="0" cellpadding="6" style="margin-top: 10px; width: 100%;">
 						<thead>
 							<tr>
-								<th class="td" scope="col"><?php echo __( 'Provider', 'woocommerce-services' ); ?></th>
-								<th class="td" scope="col"><?php echo __( 'Tracking number', 'woocommerce-services' ); ?></th>
+								<th class="td" scope="col"><?php esc_html_e( 'Provider', 'woocommerce-services' ); ?></th>
+								<th class="td" scope="col"><?php esc_html_e( 'Tracking number', 'woocommerce-services' ); ?></th>
 							</tr>
 						</thead>
 						<tbody>
-							<?php echo $markup; ?>
+							<?php echo $markup; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?> 
```

Comment:
An alternative to ignoring could be allowing restricted HTML, eg: `echo wp_kses_post( $raw_response['body'] );`

This will allow most HTML tags (all HTML tags allowed in `post` contents), but disallows risky ones such as `<script>`.

Reply:
Thanks for the suggestion. Updated to use `wp_kses_post` instead.